### DISCUSSION TITLE
Identity do sql pulando a orden

### DISCUSSION THREAD

**Discussion by joao-sotero** [balta-io/forum#177] [Tira Dúvidas] [answered]
Estou no curso de EF Core da carreira .NET.

Fiz um primeiro insert pelo banco de dados e o ID foi gerado corretamente, mas percebi que quando inseri no banco pelo codigo ele não gerou o ID na ordem certa.
![image](https://user-images.githubusercontent.com/62224433/208542987-9cd31cb7-0e61-4794-ab69-c010bf38626f.png)

**Comment by andrebaltieri** [comment, marked as answer]
Como o mundo da programação é incrível... tive este problema ontem @joao-sotero 😂
Embora no meu caso, a ordem dos IDs não me afete em nada, fiquei curioso e fui procurar.

O erro ocorre no SQL Server 2019, nas últimas versões e quando o servidor é fechado incorretamente:
https://learn.microsoft.com/en-us/answers/questions/517365/sql-server-id-jump-1000-issue-fixed-in-sql-server.html
https://blog.sqlauthority.com/2018/01/24/sql-server-identity-jumping-1000-identity_cache/

Eu tenho a péssima mania de fechar o Docker sem parar os containers, então, no meu caso acredito que tenha sido isto.

### Como resolver?
Você pode resetar o identity de qualquer tabela com `DBCC CHECKIDENT`, conforme abaixo.
```sql
DBCC CHECKIDENT ('TABELA', RESEED, 1)
```

**Comment by joao-sotero** [comment]
Eu faço exatamente isso, so desligo a maquina. Desligar um por um agora

**Reply by andrebaltieri** [reply]
😂 eu também peguei esse costume de parar todos os containers antes...
Tem um comando que ajuda:
```
docker kill $(docker ps -q)
```
